src/nyx/gpu/gabor.cu

```diff
@@ -0,0 +1,418 @@
+
+        int n[2] = {row_size, col_size};
+
+        cudaMalloc((void**)&d_image, sizeof(CuComplex)*size*batch_size);
+        if (cudaGetLastError() != cudaSuccess){
+            fprintf(stderr, "Cuda error: Failed to allocate\n");	
+        }
+
+        cudaMalloc((void**)&d_result, sizeof(CuComplex)*size*batch_size);
+        if (cudaGetLastError() != cudaSuccess){
+            fprintf(stderr, "Cuda error: Failed to allocate\n");	
+        }
+        
+        cudaMalloc((void**)&d_kernel, sizeof(CuComplex)*size*batch_size);
+        if (cudaGetLastError() != cudaSuccess){
+            fprintf(stderr, "Cuda error: Failed to allocate\n");	
+        }
+        
+        // copy data to GPU
+        cudaMemcpy(d_image, linear_image, batch_size*size*sizeof(CuComplex), cudaMemcpyHostToDevice);
+        if (cudaGetLastError() != cudaSuccess){
+            fprintf(stderr, "Cuda error: Failed to allocate\n");	
+        }
+        cudaMemcpy(d_kernel, linear_kernel, batch_size*size*sizeof(CuComplex), cudaMemcpyHostToDevice);
+        if (cudaGetLastError() != cudaSuccess){
+            fprintf(stderr, "Cuda error: Failed to allocate\n");	
+        }
+
+        cufftHandle plan;
+        cufftHandle plan_k;
+
+        int idist = size;
+        int odist = size;
+        
+        int inembed[] = {row_size, col_size};
+        int onembed[] = {row_size, col_size};
+
+        int istride = 1;
+        int ostride = 1;
+
+        if (cufftPlanMany(&plan, 2, n, inembed, istride, idist, onembed, ostride, odist, CUFFT_Z2Z, batch_size) != CUFFT_SUCCESS){
+            fprintf(stderr, "CUFFT Error: Unable to create plan\n");
+            //return;	
+        }
+        if (cufftPlanMany(&plan_k, 2, n, inembed, istride, idist, onembed, ostride, odist, CUFFT_Z2Z, batch_size) != CUFFT_SUCCESS){
+            fprintf(stderr, "CUFFT Error: Unable to create plan\n");
+            //return;	
+        }
+
+        if (cufftExecZ2Z(plan, d_image, d_image, CUFFT_FORWARD) != CUFFT_SUCCESS){
+            fprintf(stderr, "CUFFT Error: Unable to execute plan\n");
+            //return;		
+        }
+
+        if (cufftExecZ2Z(plan_k, d_kernel, d_kernel, CUFFT_FORWARD) != CUFFT_SUCCESS){
+            fprintf(stderr, "CUFFT Error: Unable to execute plan\n");
+            //return;		
+        }
+
+        if (cudaDeviceSynchronize() != cudaSuccess){
+            fprintf(stderr, "Cuda error: Failed to synchronize\n");
+            //return;
+        }
+
+        // element-wise multiplication of the image and kernel
+        cmat_mult(d_image, row_size, col_size, d_kernel, d_result, batch_size);
+
+        // transform out of fourier space
+        if (cufftExecZ2Z(plan, d_result, d_result, CUFFT_INVERSE) != CUFFT_SUCCESS){
+            fprintf(stderr, "CUFFT Error: Unable to execute plan\n");
+            //return;		
+        }
+
+        // copy results from device to host
+        cudaMemcpy(result, d_result, batch_size*size*sizeof(CuComplex), cudaMemcpyDeviceToHost); 
```

Comment:
Some of the cudaMemcpy calls have error checking after the call but some are missing. I am not sure what is the performance cost of these checks but I am in favor of having them.